### PR TITLE
Add migrateLegacyShinytest function

### PR DESCRIPTION
**Pull request by wch**
This implements `migrateLegacyShinytest`, which updates the old Shinytest directory structure to the new one.

### CODE REVIEW

File: R/test.R [hunk 1]
```diff
@@ -180,3 +190,78 @@ print.shiny_runtests <- function(x, ..., reporter = "summary") {
...
+#'  |- tests
+#'      |- shinytest.R
+#'      |- shinytest
+#'      |   `- mytest.R
+#'      |- testthat.R
+#'      `- testthat
+#'          `- test-script.R
+#' ```
+#'
+#' This allows for tests using the shinytest package as well as other testsing
```
Comment (schloerke) [inline R/test.R:217]:
```suggestion
#' This allows for tests using the shinytest package as well as other testing
```

File: R/test.R [hunk 2]
```diff
@@ -91,7 +101,7 @@ runTests <- function(
   }
 
   # See the @details section of the runTests() docs above for why this branch exists.
-  if (is_legacy_shinytest(file.path(testsDir, runners))) {
+  if (is_legacy_shinytest_dir(testsDir)) {
```
Comment (schloerke) [inline R/test.R:104]: Can we add backticks to the `?shiny::FUN` sections below?

Be something like 

```
`?shiny::migrateLegacyShinytest`
`?shiny::shinyAppTemplate`

File: R/test.R [hunk 3]
```diff
@@ -180,3 +190,78 @@ print.shiny_runtests <- function(x, ..., reporter = "summary") {
 
   invisible(x)
 }
+
+
+#' Migrate legacy shinytest files to new test directory structure
```
Comment (schloerke) [inline R/test.R:195]: Question in general... should we support documentation of package names?

https://roxygen2.r-lib.org/articles/rd-formatting.html#character-formatting

```
\pkg{package_name}
```

Comment (wch) [reply R/test.R:195]: I personally don't really have strong feelings either way. It looks like, when rendered as HTML, it just results in bold text.

**Pull request merged**